cmd/cat-main.go

```diff
@@ -270,12 +301,30 @@ func catURL(ctx context.Context, sourceURL string, encKeyDB map[string][]prefixS
 		} else {
 			return err.Trace(sourceURL)
 		}
-		gopts := GetOptions{VersionID: versionID, Zip: o.isZip, RangeStart: o.startO, PartNumber: o.partN}
-		if reader, err = getSourceStreamFromURL(ctx, sourceURL, encKeyDB, getSourceOpts{
-			GetOptions: gopts,
-			preserve:   false,
-		}); err != nil {
-			return err.Trace(sourceURL)
+
+		// Use parallel reader for multiple threads
+		if o.parallel > 1 && size > 0 && client.GetURL().Type == objectStorage {
```

Comment:
For small objects, we should just use regular operations... For example above 16MiB...
```suggestion
		if o.parallel > 1 && size > 16<<20 && client.GetURL().Type == objectStorage {
```